### PR TITLE
Jdiamond/usb logger useability

### PR DESCRIPTION
**Pull request by axlan**
* sbp_log system call (NOTE: should be removed when SBP logging is available through piksi_log interface)
* Enable/disable setting
* Clean file close on terminate signal
* piksi_log integration
* disabled by default

Addresses issues:
https://github.com/swift-nav/piksi_v3_bug_tracking/issues/391
https://github.com/swift-nav/piksi_v3_bug_tracking/issues/372
https://github.com/swift-nav/piksi_v3_bug_tracking/issues/370

This has been bench tested in a development environment, by turning the setting on and off and killing with monit

### CODE REVIEW

File: package/rotating_zmq_logger/src/rotating_zmq_logger.cc [hunk 1]
```diff
@@ -123,7 +135,52 @@ static void sigchld_handler(int signum) {
...
+  char cmd_buf[256];
+  sprintf(cmd_buf, "sbp_log --%s", log_args[priority]);
+  output = popen (cmd_buf, "w");
+  if (!output) {
+    piksi_log(LOG_ERROR, "couldn't call sbp_log.");
+    return;
+  }
+  fputs((std::string(PROGRAM_NAME) + ": " + msg_text).c_str(), output);
+  if (ferror (output)) {
+    piksi_log(LOG_ERROR, "output to sbp_log failed.");
```
Comment (jacobmcnamee) [inline package/rotating_zmq_logger/src/rotating_zmq_logger.cc]: `pclose()`

File: package/rotating_zmq_logger/src/rotating_zmq_logger.cc [hunk 2]
```diff
@@ -150,7 +209,7 @@ int main(int argc, char *argv[]) {
   sigemptyset(&sigchld_sa.sa_mask);
   sigchld_sa.sa_flags = SA_NOCLDSTOP;
   if (sigaction(SIGCHLD, &sigchld_sa, nullptr) != 0) {
-    printf("error setting up sigchld handler\n");
+    piksi_log(LOG_ERR, "error setting up sigchld handler\n");
```
Comment (jacobmcnamee) [inline package/rotating_zmq_logger/src/rotating_zmq_logger.cc]: No newlines for `piksi_log()` calls

File: package/rotating_zmq_logger/src/rotating_logger.cc [hunk 3]
```diff
@@ -38,15 +38,19 @@
 static const std::string LOG_SUFFIX = ".sbp";
 static const size_t LOG_NAME_LEN = 4 + 1 + 5 + 4;
 
-void RotatingLogger::debug_printf(const char* msg, ...) {
-  if (!_verbose_logging) {
-    return;
+static const int LOG_EMERG = 0;  /* system is unusable */
```
Comment (jacobmcnamee) [inline package/rotating_zmq_logger/src/rotating_logger.cc]: These should be defined in `<syslog.h>` which you can get from `<libpiksi/logging.h>`

**Comment by anonymous-travis-swiftnav** [comment]
## v1.1.0-branch-57-ge05221e
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.1.0-branch-57-ge05221e/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.1.0-branch-57-ge05221e/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.1.0-branch-57-ge05221e

File: package/rotating_zmq_logger/src/rotating_zmq_logger.cc (hunk 1 above)
Comment (axlan) [inline package/rotating_zmq_logger/src/rotating_zmq_logger.cc]: pclose() is called on line 159

Comment (jacobmcnamee) [inline package/rotating_zmq_logger/src/rotating_zmq_logger.cc]: Right, but that doesn't get called if you return on error here.

**Comment by axlan** [comment]
Also addresses https://github.com/swift-nav/piksi_v3_bug_tracking/issues/371

**Pull request merged**